crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -790,96 +790,61 @@ MappingPattern: ast::Pattern = {
     },
 }
 
-MatchKeywordEntry: (ast::Identifier, ast::Pattern) = {
-    <k:Identifier> "=" <v:Pattern> => (k, v),
+MatchKeywordEntry: ast::PatternKeyword = {
+    <location:@L> <attr:Identifier> "=" <pattern:Pattern> <end_location:@R> => ast::PatternKeyword {
+        attr,
+        pattern,
+        range: (location..end_location).into()
+    },
 };
 
 ClassPattern: ast::Pattern = {
-    <location:@L> <e:MatchName> "(" <patterns: OneOrMore<Pattern>> "," <kwds:OneOrMore<MatchKeywordEntry>> ","? ")" <end_location:@R> => {
-        let (kwd_attrs, kwd_patterns) = kwds
-            .into_iter()
-            .unzip();
-        ast::PatternMatchClass {
-            cls: Box::new(e),
-            patterns,
-            kwd_attrs,
-            kwd_patterns,
-            range: (location..end_location).into()
-        }.into()
-    },
-    <location:@L> <e:MatchName> "(" <patterns: OneOrMore<Pattern>> ","? ")" <end_location:@R> => {
-        ast::PatternMatchClass {
-            cls: Box::new(e),
-            patterns,
-            kwd_attrs: vec![],
-            kwd_patterns: vec![],
-            range: (location..end_location).into()
-        }.into()
-    },
-    <location:@L> <e:MatchName> "(" <kwds:OneOrMore<MatchKeywordEntry>> ","? ")" <end_location:@R> => {
-         let (kwd_attrs, kwd_patterns) = kwds
-            .into_iter()
-            .unzip();
+    <location:@L> <cls:MatchName> <arguments:ClassArguments> <end_location:@R> => {
         ast::PatternMatchClass {
-            cls: Box::new(e),
-            patterns: vec![],
-            kwd_attrs,
-            kwd_patterns,
+            cls: Box::new(cls),
+            arguments,
             range: (location..end_location).into()
         }.into()
     },
-    <location:@L> <e:MatchName> "(" ")" <end_location:@R> => {
+    <location:@L> <cls:MatchNameOrAttr> <arguments:ClassArguments> <end_location:@R> => {
         ast::PatternMatchClass {
-            cls: Box::new(e),
-            patterns: vec![],
-            kwd_attrs: vec![],
-            kwd_patterns: vec![],
+            cls: Box::new(cls),
+            arguments,
             range: (location..end_location).into()
         }.into()
     },
-    <location:@L> <e:MatchNameOrAttr> "(" <patterns: OneOrMore<Pattern>> "," <kwds:OneOrMore<MatchKeywordEntry>> ","? ")" <end_location:@R> => {
-        let (kwd_attrs, kwd_patterns) = kwds
-            .into_iter()
-            .unzip();
-        ast::PatternMatchClass {
-            cls: Box::new(e),
+}
+
+ClassArguments: ast::PatternArguments = {
```

Comment:
Nit: I got really confused by `ClassArguments`. I first thought you have two rules, one for `ClassArguments` and one for `PatternArguments`. I recommend renaming the rule to the same name as the node as we do for other rules too